design/status-design-update.md

```diff
@@ -0,0 +1,1166 @@
+- `description` performs the same function as the global `description` does right now, just closer to the source of the problem.
+
+So, a generic `status` stanza for each type of thing below would look like:
+
+```yaml
+status:
+  currentStatus: <Valid|Invalid|Warning>
+  description: <text>
+```
+
+#### `status.includes`
+
+This field should show the status of each include in the `spec.includes` array.
+In order to fully identify them, this means that this must include all the information in each `spec.includes` item, plus the general `status` stanza.
+
+That is, this field should include status of the includes *in this HTTPPRoxy*.
+
+See the examples below to see this in action.
+
+#### `status.routes`
+
+Similarly to `status.includes`, this should include the information from each `route` stanza in `spec.routes`, with an added `status` stanza.
+
+This field should include status of the routes in this HTTPProxy.
+
+#### `status.appliedConditions`
+
+This is where this gets trickier.
+This field is intended to show what conditions are applied when this HTTPProxy object is included from somewhere.
+
+In order to do this, the following information must be present, somehow:
+
+- full list of merged conditions. That is, all prefixes  must be fully merged, and the full set of headers must be visible.
+- a unique identifier for the condition set. In practice, this will end up being related to the inclusion path to this HTTPProxy.
+
+The first is relatively straightforward, a `conditions` array, with the merge rules having been run on it, will suffice.
+
+The second is important for higher-order tool use as well as service admin use.
+
+The exact implementation here will depend on if we do the optional field I include below.
+
+In any case, there should be an `includePath` field in the `status.appliedConditions[n]` object that ensures uniqueness.
+
+This could be rendered as a string, representing the include path, preferably including the FQDN of the root.
+
+So, something that looked like this:
+
+```yaml
+status:
+  appliedConditions:
+    - conditions:
+      - prefix: /foo/bar/baz
+      includePath: "root:www.foobar.com,foo,bar"
+```
+
+Alternatively, if we are not including a more verbose representation of the computed DAG in each object, something like this would be required:
+
+```yaml
+status:
+  appliedConditions:
+    - conditions:
+      - prefix: /foo/bar/baz
+      includePath:
+        name: root
+        namespace: default
+        fqdn: www.foobar.com
+        includes:
+          name: foo
+          namespace: default
+          conditions:
+          - prefix: /foo
+          includes:
+            name: bar
+            namespace: default
+            conditions:
+            - prefix: /bar
+            includes:
+              name: baz
+              namespace: default
+              conditions:
+              - prefix: /baz
+```
+
+#### Optional: `status.proxyGraph`
```

Comment:
To me, this information isn't `Status`. Maybe something else, or I just haven't gotten my head around it enough.

Reply:
It's the status of the computed HTTPProxy tree.

Spec contains the declarative definition of what an object should look like, status contains anything computed, like Load Balancer IPs. In this case, the proxyGraph is computed by the superset of all HTTPProxy objects.

I'm not sure, to be honest, if we're better having this full-data construct, or being able to reconstruct it from the other three fields.

Reply:
What's the use case for this information? I don't think it should be optional because we're either going to have to describe it fully in this design, then not implement it, or the opposite, underspecify it in this document and maybe implement it, sorta. I'd suggest either making it a concrete deliverable of this document or moving it out of scope.